zbuf/batch.go

```diff
@@ -143,6 +143,7 @@ func (r *pullerReader) Read() (*zed.Value, error) {
 	rec := &vals[r.idx]
 	r.idx++
 	if r.idx == len(vals) {
+		r.batch.Unref()
```

Comment:
Once `r.batch.Unref` is called, the storage referenced by both `rec` and `rec.Bytes` is subject to immediate reuse, so we need to copy both. We could do that with `rec = rec.Copy()` here, but it's more efficient to add a `val zed.Value` field to `pullerReader` and do this:
```suggestion
		r.val.CopyFrom(rec)
		rec = &r.val
		r.batch.Unref()
```